db/collection.go

```diff
@@ -151,53 +123,41 @@ func (db *db) createCollection(
 		return nil, err
 	}
 	desc.ID = uint32(colID)
-	col, err := db.newCollection(desc)
-	if err != nil {
-		return nil, err
-	}
 
-	// Local elements such as secondary indexes should be excluded
-	// from the (global) schemaId.
-	schemaBuf, err := json.Marshal(col.desc.Schema)
+	col, err := db.newCollection(desc, schema)
 	if err != nil {
 		return nil, err
 	}
 
-	// add a reference to this DB by desc hash
-	cid, err := ccid.NewSHA256CidV1(schemaBuf)
+	schema, err = descriptions.CreateSchemaVersion(ctx, txn, schema)
 	if err != nil {
 		return nil, err
 	}
-	schemaID := cid.String()
-	col.schemaID = schemaID
-
-	// For new schemas the initial version id will match the schema id
-	schemaVersionID := schemaID
 
-	col.desc.Schema.VersionID = schemaVersionID
-	col.desc.Schema.SchemaID = schemaID
+	col.schema = schema
+	col.desc.SchemaVersionID = schema.VersionID
 
 	// buffer must include all the ids, as it is saved and loaded from the store later.
 	buf, err := json.Marshal(col.desc)
 	if err != nil {
 		return nil, err
 	}
 
-	collectionSchemaVersionKey := core.NewCollectionSchemaVersionKey(schemaVersionID)
+	collectionSchemaVersionKey := core.NewCollectionSchemaVersionKey(schema.VersionID)
 	// Whilst the schemaVersionKey is global, the data persisted at the key's location
 	// is local to the node (the global only elements are not useful beyond key generation).
 	err = txn.Systemstore().Put(ctx, collectionSchemaVersionKey.ToDS(), buf)
 	if err != nil {
 		return nil, err
 	}
 
-	collectionSchemaKey := core.NewCollectionSchemaKey(schemaID)
-	err = txn.Systemstore().Put(ctx, collectionSchemaKey.ToDS(), []byte(schemaVersionID))
+	collectionSchemaKey := core.NewCollectionSchemaKey(schema.SchemaID)
+	err = txn.Systemstore().Put(ctx, collectionSchemaKey.ToDS(), []byte(schema.VersionID))
 	if err != nil {
 		return nil, err
 	}
 
-	err = txn.Systemstore().Put(ctx, collectionKey.ToDS(), []byte(schemaVersionID))
+	err = txn.Systemstore().Put(ctx, collectionKey.ToDS(), []byte(schema.VersionID))
```

Comment:
thought: These make no sense anymore. I think this is what you were referring to [here](https://github.com/sourcenetwork/defradb/issues/1964)

Reply:
Yes :)